### PR TITLE
Rate limit tracker popups to 5 per hour

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

**Comment by maxtaco** [comment]
As I'm reading this code, I see a potential issue: if I access `max,alice` but my track of `alice` failed, I might not see a popup window if I've chewed through my rate limit. So I'll just see an I/O failure or some such without any further explanation.

Checking this with @malgorithms  and cc: @strib 

@strib: the context is that we're trying to workaround the popup window storm that happens to me (and also mike) when finder starts rooting around in KBFS and accesses all favorited folders.

File: go/engine/buffered_identify_ui.go [hunk 1]
```diff
@@ -77,12 +78,18 @@ func (b *bufferedIdentifyUI) flush(trackingBroke bool) (err error) {
 		return nil
 	}
 
+	if !b.G().IdentifyUILimiter.Wait(5 * time.Millisecond) {
```
Comment (maxtaco) [inline go/engine/buffered_identify_ui.go]: Maybe the solution to my concern is no to allow ratelimiting if you track them --- and to only use it if you don't...

File: go/libkb/burstlimit.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package libkb
+
+import "time"
+
+// BurstLimiter is a rate limiter that allows bursts of requests.
+type BurstLimiter struct {
```
Comment (maxtaco) [inline go/libkb/burstlimit.go]: did Go's rate limiter not cut it?

File: go/engine/identify2_with_uid.go [hunk 3]
```diff
@@ -709,13 +709,34 @@ func (e *Identify2WithUID) runIdentifyUI(netContext context.Context, ctx *Contex
...
+		}
+		if e.arg.Reason.IsUIProfile() {
+			// identifies from the UI for user profiles use DEFAULT_KBFS, so check
+			// the reason and don't rate limit if it is a UI profile identify.
+			checkRateLimit = false
+		}
+	}
+
+	if checkRateLimit {
+		if !e.G().IdentifyUILimiter.Wait(5 * time.Millisecond) {
```
Comment (maxtaco) [inline go/engine/identify2_with_uid.go]: Can we move this constant Wait into the library?

**Comment by strib** [comment]
If the service returns an error to KBFS on identify, then yes KBFS will return an EIO.

**Comment by patrickxb** [comment]
Once it hits the standard identify UI code path, it's very difficult to determine which call will end up bringing up the tracker window.  

There are a lot of conditions being checked in the UI and 8 different places where the tracker popup could be displayed.

**Comment by maxtaco** [comment]
How about this rule: you can't rate-limit if you track them...

**Comment by patrickxb** [comment]
Also, KBFS won't get an error.  It will just be a suppressed tracker popup:

                        e.G().Log.Debug("| identify ui rate limit exceeded, suppressing tracker popups")
                        iui = newLoopbackIdentifyUI(e.G(), &e.trackBreaks)

I'm not saying that's correct, just that KBFS won't get an error.

**Comment by patrickxb** [comment]
@maxtaco would that be helpful, though?  won't you still get a popup window storm if, for example, github proof checks are failing?

**Comment by maxtaco** [comment]
Right, I understand that... I don't follow your earlier point about the standard identify UI.  I agree with what you write, but I don't see why that's relevant here.

**Comment by patrickxb** [comment]
In the buffered identify UI, there is the `flush()` function where we can tell that it will be spawning a tracker popup.  

But in the "standard" identify UI, we don't know which UI call will be the one that spawns the tracker popup.

**Comment by maxtaco** [comment]
right, I agree with all of that too...

File: go/engine/buffered_identify_ui.go (hunk 1 above)
Comment (maxtaco) [reply go/engine/buffered_identify_ui.go]: In other words, @patrickxb, my proposal is to delete these 3 lines...

File: go/libkb/burstlimit.go (hunk 2 above)
Comment (patrickxb) [reply go/libkb/burstlimit.go]: It seemed like overkill and tuned for much speedier rates.

**Comment by maxtaco** [comment]
yeah, i would still be hosed in the github outage case....

i guess i'll have to live with that until the underlying KBFS bug is fixed...

**Comment by maxtaco** [comment]
cc @strib can you remind me which KBFS ticket has this issue? thx...

File: go/engine/buffered_identify_ui.go (hunk 1 above)
Comment (patrickxb) [reply go/engine/buffered_identify_ui.go]: Ok.

**Comment by maxtaco** [comment]
i dunno @patrickxb, I guess we can't get this perfect

**Comment by patrickxb** [comment]
I'll remove it from buffered...that seems like a fine solution for now.

**Comment by strib** [comment]
There's no underlying KBFS bug.  This is an OS/app thing.  We can special case certain types of lookups (like the ".localized" thing discussed in https://keybase.atlassian.net/browse/KBFS-2649), maybe that's what you're referring to?

**Comment by maxtaco** [comment]
Thanks @strib that's the one.  Right, I think we should make an exception for `.localized`. I think @mmaxim and @malgorithms have also hit this issue.

**Comment by maxtaco** [comment]
I was hoping we could solve on the core side, but it looks like it's fundamentally not resolvable. 

Hmm, one workaround might be for KBFS to pipe through a "canRateLimit" flag on accesses such as these. If we get that flag and we track, then we still can use the rate limiter introduced here

**Comment by patrickxb** [comment]
@maxtaco I pushed changes based on your feedback when you get a chance.

**Pull request merged**